File: app/src/main/kotlin/com/mozzid/data/local/DetectionDao.kt
package com.mozzid.data.local

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface DetectionDao {
    @Query("SELECT * FROM detections ORDER BY timestamp DESC")
    fun watchAll(): Flow<List<DetectionEntity>>

    @Query("SELECT * FROM detections ORDER BY timestamp DESC")
    suspend fun all(): List<DetectionEntity>

    @Insert
    suspend fun insert(entity: DetectionEntity): Long

    @Query("DELETE FROM detections WHERE id = :id")
    suspend fun delete(id: Long)

    @Query("DELETE FROM detections")
    suspend fun deleteAll()

    @Query("SELECT COUNT(*) FROM detections")
    suspend fun count(): Int
}

File: reference/flutter-src/lib/data/local/detection_dao.dart
import 'package:sqflite/sqflite.dart';

import '../../domain/models/detection.dart';

/// Thin data-access object over the `detections` table. Raw SQL, no codegen.
class DetectionDao {
  DetectionDao(this._db);

  final Database _db;

  Future<List<Detection>> all() async {
    final rows = await _db.query('detections', orderBy: 'timestamp DESC');
    return rows.map(Detection.fromMap).toList();
  }

  Future<Detection> insert(Detection detection) async {
    final id = await _db.insert('detections', detection.toMap());
    return detection.copyWith(id: id);
  }

  Future<void> delete(int id) =>
      _db.delete('detections', where: 'id = ?', whereArgs: [id]);

  Future<void> deleteAll() => _db.delete('detections');
}
